src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+   - After a run is finished, the same run handle can be reused to execute the kernel multiple times if desired. 
+     
+       - API ``xrtRunSetArg`` is used to set one or more arguments, in the example above only the last (3rd) argument is changed before the second execution
+       - API ``xrtRunStart`` is used to execute the kernel using the run handle. 
+   - API ``xrtRunClose`` is used to close the Run handle.  
+ 
+   
+**C++**: The equivalent C++ code
+
+In C++ the ``xrt::kernel`` class provides **overloaded operator ()** to execute the kernel with a comma-separated list of arguments.  
+
+.. code:: c++
+      :number-lines: 50
+   
+           // 1st kernel execution
+           auto run = kernel(buf_a, buf_b, scalar_1); 
+           run.wait();
+    
+           // 2nd kernel execution with just changing 3rd argument
+           run.set_arg(2,scalar_2); // Arguments are specified starting from 0 
+           run.start();
+           run.wait();
+
+The above c++ code block is demonstrating 
+  
+  - The kernel execution using the ``xrt::kernel()`` operator with the list of arguments that returns a xrt::run object. This is an asynchronous API and returns after submitting the task.    
+  - The method ``xrt::run::wait`` is used to block the current thread until the current execution is finished. 
+  - The method ``xrt::run::set_arg`` is used to set one or more kernel argument(s) before the next execution. In the example above, only the last (3rd) argument is changed.  
+  - The method ``xrt::run::start`` is used to start the next kernel execution with new argument(s).   
+
+Other kernel execution related APIs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Run handle/object supports few other use-cases. 
+
+**Obtaining the run handle/object before execution**: In the above example we have seen a run handle/object is obtained when the kernel is executed (kernel execution returns a run handle/object). However, a run handle can be obtained even before the kernel execution. The flow is as below
+
+    - Open a Run handle (or object) by API ``xrtRunOpen`` (in C++ ``xrt::run`` constructor with a kernel argument). There is no kernel execution associated with this run handle/object yet
+    - Set the kernel arguments associated for the next execution by ``xrtRunSetArg`` (in C++ method ``xrt::run::set_arg``). 
+    - Execute the kernel by ``xrtRunStart`` (in C++ method ``xrt::run::start``).
+    - Wait for the execution finish by ``xrtRunWait`` (C++: ``xrt::run::wait``). 
+
+**Timeout while wait for kernel finish**: The API ``xrtRunWait`` blocks the current thread until the kernel execution finishes. However, a timeout supported API ``xrtRunWaitFor`` is also provided . The timeout number can be specified using a millisecond unit.
+
+In C++, the timeout facility can be used by the same ``xrt::run::wait(unsigned int timeout_ms=0)`` method by providing a millisecond number as an argument. 
```

Comment:
Use `std::chrono` instead of random units

Reply:
implementation-specific. I am tracking this trough a CR. 
Nothing to do for this PR, hence resolving the conversation.